### PR TITLE
Allow to edit mapping for categorical scale only

### PR DESCRIPTION
**Pull request by zBritva**

### CODE REVIEW

File: src/app/views/panels/scale_editor.tsx [hunk 1]
```diff
@@ -76,46 +82,52 @@ export class ScaleEditor extends React.Component<
...
+              {canExtendLegend ? (
+                <>
+                  <ButtonRaised
+                    url={R.getSVGIcon("general/plus")}
+                    text={strings.scaleEditor.add}
+                    onClick={() => {
+                      const mappingsKey = Object.keys(scale.properties.mapping);
+                      const theLastMapping: any = mappingsKey[
+                        mappingsKey.length - 1
+                      ] as any;
```
Comment (danmarshall) [inline src/app/views/panels/scale_editor.tsx:94]: Why cast as `any`?

**Review by danmarshall — CHANGES_REQUESTED** [review]

File: src/app/views/panels/scale_editor.tsx [hunk 2]
```diff
@@ -89,12 +89,11 @@ export class ScaleEditor extends React.Component<
                     text={strings.scaleEditor.add}
                     onClick={() => {
                       const mappingsKey = Object.keys(scale.properties.mapping);
-                      const theLastMapping: any = mappingsKey[
-                        mappingsKey.length - 1
-                      ] as any;
+                      const theLastMapping: string =
+                        mappingsKey[mappingsKey.length - 1];
                       const value = (scale.properties.mapping as any)[
```
Comment (danmarshall) [inline src/app/views/panels/scale_editor.tsx:94]: Also another `any` here, perhaps you can cast to `unknown` and then to the type you want.

**Review by danmarshall — CHANGES_REQUESTED** [review]

File: src/app/views/panels/scale_editor.tsx (hunk 2 above)
Comment (zBritva) [reply src/app/views/panels/scale_editor.tsx:94]: Cast to AttributeMap

File: src/app/views/panels/scale_editor.tsx [hunk 3]
```diff
@@ -76,46 +83,51 @@ export class ScaleEditor extends React.Component<
...
+                    }}
+                  />
+                  <ButtonRaised
+                    url={R.getSVGIcon("general/minus")}
+                    text={strings.scaleEditor.removeLast}
+                    onClick={() => {
+                      const mappingsKey = Object.keys(scale.properties.mapping);
+                      const theLastMapping: string = mappingsKey[
+                        mappingsKey.length - 1
+                      ] as string;
```
Comment (danmarshall) [inline src/app/views/panels/scale_editor.tsx:115]: remove `as string`

**Review by danmarshall — APPROVED** [review]

**Pull request merged**